osu.Server.Queues.ScoreStatisticsProcessor/Processors/MedalAwarders/StarRatingMedalAwarder.cs

```diff
@@ -0,0 +1,179 @@
+﻿using JetBrains.Annotations;
+using MySqlConnector;
+using osu.Game.Beatmaps;
+using osu.Game.Online.API.Requests.Responses;
+using osu.Game.Rulesets;
+using osu.Game.Rulesets.Difficulty;
+using osu.Game.Rulesets.Mania.Mods;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Scoring;
+using osu.Server.Queues.ScoreStatisticsProcessor.Helpers;
+using osu.Server.Queues.ScoreStatisticsProcessor.Models;
+using osu.Server.Queues.ScoreStatisticsProcessor.Stores;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Beatmap = osu.Server.Queues.ScoreStatisticsProcessor.Models.Beatmap;
+
+namespace osu.Server.Queues.ScoreStatisticsProcessor.Processors.MedalAwarders
+{
+    [UsedImplicitly]
+    public class StarRatingMedalAwarder : IMedalAwarder
+    {
+        private BeatmapStore? beatmapStore;
+
+        public bool RunOnFailedScores => false;
+
+        public IEnumerable<Medal> Check(SoloScoreInfo score, UserStats userStats, IEnumerable<Medal> medals, MySqlConnection conn, MySqlTransaction transaction)
+        {
+            return checkAsync(score, medals, conn, transaction).Result;
+        }
+
+        private async Task<IEnumerable<Medal>> checkAsync(SoloScoreInfo score, IEnumerable<Medal> medals, MySqlConnection conn, MySqlTransaction transaction)
+        {
+            List<Medal> earnedMedals = new List<Medal>();
+
+            Ruleset ruleset = LegacyRulesetHelper.GetRulesetFromLegacyId(score.RulesetID);
+            Mod[] mods = score.Mods.Select(m => m.ToMod(ruleset)).ToArray();
+
+            // Ensure the score isn't using any difficulty reducing mods
+            if (mods.Any(x => x.IsDifficultyReductionMod()))
+                return Enumerable.Empty<Medal>();
+
+            // On mania, these medals cannot be triggered with key mods and Dual Stages
+            if (score.RulesetID == 3 && mods.Any(isManiaDisallowedMod))
+                return Enumerable.Empty<Medal>();
+
+            try
+            {
+                beatmapStore ??= await BeatmapStore.CreateAsync(conn, transaction);
+
+                Beatmap? beatmap = await beatmapStore.GetBeatmapAsync(score.BeatmapID, conn, transaction);
+                if (beatmap == null)
+                    return Enumerable.Empty<Medal>();
+
+                // Make sure the map isn't Qualified or Loved, as those maps may occasionally have SR-breaking/aspire aspects
+                if (beatmap.approved == BeatmapOnlineStatus.Qualified || beatmap.approved == BeatmapOnlineStatus.Loved)
+                    return Enumerable.Empty<Medal>();
```

Comment:
i'm not sure i get this logic. aren't graveyarded maps also prone to sr-breaking aspects?

Reply:
Anything that isn't ranked should not be giving any medals period. Likely a general oversight at a higher level.

Reply:
@Walavouchey addressed in 85137b7affa0251115d0e907db5b3cae84616e39 (with test coverage in f22be2b2b9df6c81e8a83758bc79eef309864f32).

There is _one_ medal that could at a time be granted on unranked beatmaps. That was this December, and it's now turned off to the best of my knowledge. We don't talk about this one, _wink wink_. (And even if we were to, it wouldn't be implemented in this public reposiory.)